activerecord/lib/active_record/railtie.rb

```diff
@@ -133,7 +133,10 @@ class Railtie < Rails::Railtie # :nodoc:
               next if current_version.nil?
 
               cache = YAML.load(File.read(filename))
-              if cache.version == current_version
+              case
+              when cache.missing_indexes?
+                warn "Ignoring db/schema_cache.yml because it is missing index information. Generate a new schema cache with indexes."
```

Comment:
Should we ignore the schema cache in this case? If it is missing the indexes it is because it was generated in a version that didn't know how to dump it. This just means that this app will now make queries to get the index. With this check this app will make queries to all database information as opposite to just indexes.